test/functional/test-viewer.js

```diff
@@ -500,28 +500,47 @@ describe('Viewer', () => {
     expect(viewer.messageQueue_[0].eventType).to.equal('documentLoaded');
   });
 
-  it('should not request cid', () => {
-    sandbox.stub(viewer, 'isTrustedViewer', () => {
-      return Promise.resolve(false);
+  describe.only('baseCid', () => {
+    const cidData = JSON.stringify({
+      time: 0,
+      cid: 'cid-123',
     });
-    sandbox.stub(viewer, 'sendMessage', () => {
-      return Promise.reject('should not be requested');
+    let trustedViewer;
+    let persistedCidData;
+
+    beforeEach(() => {
+      trustedViewer = true;
+      persistedCidData = cidData;
+      sandbox.stub(viewer, 'isTrustedViewer',
+          () => Promise.resolve(trustedViewer));
+      sandbox.stub(viewer, 'sendMessage', (message, payload) => {
+        if (message != 'cid') {
+          return Promise.reject();
+        }
+        if (payload) {
+          persistedCidData = payload;
+        }
+        return Promise.resolve(persistedCidData);
+      });
     });
-    return viewer.getBaseCid().then(cid => {
-      expect(cid).to.be.undefined;
+
+    it('should return CID', () => {
+      expect(viewer.baseCid()).to.eventually.equal(cidData);
     });
-  });
 
-  it('should request cid for trusted viewer', () => {
-    sandbox.stub(viewer, 'isTrustedViewer', () => {
-      return Promise.resolve(true);
+    it('should not request cid for untrusted viewer', () => {
+      trustedViewer = false;
+      expect(viewer.baseCid()).to.eventually.be.undefined;
     });
-    sandbox.stub(viewer, 'sendMessage', name => {
-      expect(name).to.equal('cid');
-      return Promise.resolve('from-viewer');
+
+    it('should convert CID returned by legacy API to new format', () => {
+      persistedCidData = 'cid-123';
+      expect(viewer.baseCid()).to.eventually.equal(cidData);
```

Comment:
How does it get time value of `0`? I don't see fake clock anywhere here. In either case, is there a way to test against, e.g. `time = 1`? `0` is too "accidental".